src/include/txn_inline.h

```diff
@@ -1194,6 +1194,19 @@ __wt_txn_id_check(WT_SESSION_IMPL *session)
     if (F_ISSET(txn, WT_TXN_HAS_ID))
         return (0);
 
+    /*
+     * Return error when the transactions with read committed or uncommitted isolation tries to
+     * perform any write operation. Skip metadata because metadata searches (like schema operations)
+     * rely on WT_ISO_READ_UNCOMMITTED and metadata checkpoints rely on WT_ISO_READ_COMMITTED
+     * isolation.
+     */
+    if (session->dhandle != NULL && !WT_IS_METADATA(session->dhandle) &&
+      (txn->isolation == WT_ISO_READ_COMMITTED || txn->isolation == WT_ISO_READ_UNCOMMITTED)) {
+        WT_ASSERT(session, !F_ISSET(session, WT_SESSION_INTERNAL));
+        WT_RET_MSG(
+          session, ENOTSUP, "not supported in read-committed or read-uncommitted transactions.");
```

Comment:
I feel the error message should be more clear as `write operations are not ...`?

Reply:
Yeah, this sentence is more relevant. Updated the comments